cirq-core/cirq/protocols/json_serialization.py

```diff
@@ -284,14 +284,16 @@ def _json_dict_with_cirq_type(obj: Any):
         # TODO: upgrade to ValueError in v0.15
         warnings.warn(
             f"Found 'cirq_type': '{base_dict['cirq_type']}' in user-specified _json_dict_. "
-            "This field is now automatically generated from the class name and (if present) "
-            "its _json_namespace_ method; this produces the same serialization format as "
-            "the existing convention of `[namespace.]class_name`."
+            "'cirq_type' is now automatically generated from the class's name and its "
+            "_json_namespace_ method as `cirq_type: '[<namespace>.]<class_name>'`."
             "\n\n"
             "Starting in v0.15, custom 'cirq_type' values will trigger an error. "
-            "To fix this, remove 'cirq_type' from the class _json_dict_ method. Classes "
-            "outside of cirq should also define a non-empty _json_namespace_ to avoid "
-            "collision with Cirq classes.",
+            "To fix this, remove 'cirq_type' from the class _json_dict_ method and "
+            "define _json_namespace_ for the class."
+            "\n\n"
+            "For backwards compatibility, classes whose old 'cirq_type' does not match the "
+            "new value must appear in `json_resolver_cache.py::_class_resolver_dictionary()` "
```

Comment:
I don't understand this instruction. My project does not have a `json_resolver_cache.py` file. Is this a file within cirq? Say it is within cirq, and give the full file path not the name.

Also, this seems rather extreme to require someone who wrote a different library to make a change in cirq to maintaing compatibility across versions? It sounds like this is saying I *can't* maintain compatibility.

Reply:
Sorry, I misunderstood how Cirq handles third-party JSON resolvers. The appropriate thing to update for this is whatever custom resolver you are using, as described [here](https://github.com/quantumlib/Cirq/blob/efd3b25e83a6dacde69d7d000b7de19d8671cea8/cirq-core/cirq/protocols/json_serialization.py#L73). I'll update the instructions to match.